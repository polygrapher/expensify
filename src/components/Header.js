import React from 'react';
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
import {bindActionCreators} from 'redux';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">New Expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        <button onClick={startLogout}>Log out</button>
    </header>
);

const mapDispatchToProps = (dispatch) => bindActionCreators({startLogout}, dispatch);

export default connect(undefined, mapDispatchToProps)(Header);