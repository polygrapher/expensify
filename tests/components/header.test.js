import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../src/components/Header';

test('Header renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot('Expensify');
});
