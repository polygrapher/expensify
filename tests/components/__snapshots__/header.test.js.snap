// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly: Expensify 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    New Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/edit"
  >
    Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
  <button>
    Log out
  </button>
</header>
`;
